sklearn/cluster/tests/test_bicluster.py

```diff
@@ -210,6 +210,7 @@ def test_perfect_checkerboard():
 @pytest.mark.parametrize(
     "args",
     [
+        {"n_init": 0},
```

Comment:
In the `pytest.raises` context manager, we need to match the error message with `match=err_msg`. `err_msg` will be in the parametrisation.

Reply:
We should as well check that we trigger the `TypeError` by passing another type than the one expected.